refactor(FR-357): NEO configurations page using Setting List

nowgnuesLee opened the pull request:
resolves #3015 [(FR-357)](https://lablup.atlassian.net/browse/FR-357)

**changes**
* Changed configurations items into react `SettingList` component
* Update modal design to current design system. Enable to auto-save and delete fuctionality.
* Changed `notification` to `message`
* add `Save` button and validate.
* add checkbox to unset a specific value
* add `note` component on upside.
<img width="262" alt="스크린샷 2025-02-17 오후 2 48 01" src="https://github.com/user-attachments/assets/296a6201-7019-4b28-ad7e-8ae18ebdf78d" />

![스크린샷 2025-03-17 오전 11.28.56.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/xbCemO1RqqcSjEXCRK3p/dec1a765-0d34-49fb-9ea6-5bad22a9cd5a.png)

![스크린샷 2025-03-26 오후 5.27.53.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/xbCemO1RqqcSjEXCRK3p/28e28945-e066-4898-8d9a-18735552f115.png)

**Checklist:** (if applicable)

- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

agatha197 reviewed (changes requested):
Please resolve the conflicts. And the margin between the text and the button is too narrow. Let's enlarge it.
<img width="758" alt="image" src="https://github.com/user-attachments/assets/d409e55d-6aa7-4606-be19-a2643db5f1a2" />

lizable reviewed (changes requested):
I have a several requests to change before this PR is merged.

1. Update on-the-fly 
I'm not sure update on blur event is a good choice or not.
In both "overlay network" setting and "scheduler". I think is better to have  "update" button to click so that they know what they've been changed, explicitly. Besides, everytime input field is on blur, update happens; which makes continuous update requests with snackbar, which seems a bit distracted. Therefore I guess It's better to have "update" button in the modal for user to request explicitly. If you have any other options, please tell us.
[Screen Recording 2025-02-13 at 5.44.59 PM.mov <span class="graphite__hidden">(uploaded via Graphite)</span> <img class="graphite__hidden" src="https://app.graphite.dev/api/v1/graphite/video/thumbnail/Dgoz5XqHffJdivyccymr/5df899dc-9e33-4165-8656-806ec3103d37.mov" />](https://app.graphite.dev/media/video/Dgoz5XqHffJdivyccymr/5df899dc-9e33-4165-8656-806ec3103d37.mov)

2. Would Snackbar notification is the best option?
I'm not sure about this either...probably after updating the request from above will make it okay, but I think we should make the height of notification(snack bar) little bit shortened than current status. I think is too big to distract the changes.

nowgnuesLee commented:
1. 
I think so. Previously, a warning modal would appear when user doesn't save the value and close it, but I removed that modal and changed it to save on-the-fly using `onBlur` (related [thread](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1738570209920?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1738570209920&teamName=devops&channelName=Frontend&createdTime=1738570209920). What do you think about displaying a warning message above instead of the warning modal? I think it would be better. After that let's add update button.

2. 
I used that notification due to the previous legacy. How about using the `message` feature from App.useApp() instead?

yomybaby commented on react/src/App.tsx:
```diff
@@ -354,7 +357,11 @@ const router = createBrowserRouter([
       },
       {
         path: '/settings',
-        handle: { labelKey: 'webui.menu.Configurations' },
```
Please recover handle. It's for display breadcrumb title.

nowgnuesLee replied:
Oh, I'm sorry. I'll recover it.

yomybaby commented on react/src/hooks/useBAIConfigurationsSetting.ts:
```diff
@@ -0,0 +1,223 @@
+      );
+      if (result === 'ok') {
+        updatePulling();
+        message.success(t('notification.SuccessfullyUpdated'));
```
It's not a good idea to call a function to render the UI (e.g., `message.success`). My suggestion is to remove `useBAIConfigurationSettings` and move all the logic from this file to each item in `ConfigurationSettingList.tsx`. Since the logic for each item is neither lengthy nor dependent, it would be better to keep the related code close together to enhance readability.

nowgnuesLee replied:
I had been thinking about it as well, but your suggestion has made it clearer. Thank you.

yomybaby commented on react/src/components/SettingItem.tsx:
```diff
@@ -74,7 +78,12 @@ const SettingItem: React.FC<SettingItemProps> = ({
-            {...selectProps}
+            disabled={disabled}
+            style={{
+              ...selectProps?.style,
+              marginTop: token.marginXS,
```
The style from the prop should have higher priority. More specifically defined styles should take precedence.

nowgnuesLee replied:
Thank you! I'll fix it up.

lizable reviewed (dismissed):
@nowgnuesLee  It would be nice to fix the conflict so that I can review this :)

yomybaby commented on react/src/components/OverlayNetworkSettingModal.tsx:
```diff
@@ -0,0 +1,125 @@
+              <Flex>
+                {labels[key]}
+                <Tooltip title={tooltipText[key]}>
+                  <Button type="link" icon={<InfoCircleOutlined />} />
```
Form.Item has 'tooltip` prop. Please use it. :)

nowgnuesLee replied:
Oh, I didn't know about it. Thank you for pointing that out. :)

yomybaby commented on react/src/components/OverlayNetworkSettingModal.tsx:
```diff
@@ -0,0 +1,125 @@
+    >
+      <Form ref={formRef} initialValues={networkOptions}>
+        {Object.keys(networkOptions).map((key) => (
+          <Form.Item
+            label={
+              <Flex>
+                {labels[key]}
+                <Tooltip title={tooltipText[key]}>
+                  <Button type="link" icon={<InfoCircleOutlined />} />
+                </Tooltip>
+              </Flex>
+            }
+            required
+            rules={[
+              {
+                required: true,
+                message: t('data.explorer.ValueRequired'),
+              },
+            ]}
+            name={key}
+            key={key}
+          >
+            <InputNumber
+              min={optionRange[key].min}
+              max={optionRange[key].max}
+              style={{
+                width: '100%',
+              }}
+            />
+          </Form.Item>
```
It doesn't seem necessary to manage `networkOptions` as a separate array and use the map function to create the UI. It looks like you did this considering the possibility of adding more options in the future. However, currently, there's only one option, and there's no guarantee that future additions will follow the same rule or use `InputNumber`.

nowgnuesLee replied:
I agree. I will modify everything that was managed as an array. Thank you! :)

nowgnuesLee commented:
* **#3077** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/3077?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/3077?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>flow:merge-queue</kbd> - adds this PR to the back of the merge queue
- <kbd>flow:hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/lablup?ref=merge-queue-instructions-in-stack-comment&prId=3077).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

yomybaby reviewed (changes requested):
The previous version and the new version display different values. Please double-check. I checked it in dogbowl.

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/XqC2uNFuj0wg8I60sMUh/476d02e5-18bd-43b8-8950-c7eb84dc0ca4.png)

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/XqC2uNFuj0wg8I60sMUh/6a7f7af6-aa86-4e58-b26f-6c156b8db018.png)

agatha197 commented on react/src/components/ConfigurationsSettingList.tsx:
```diff
@@ -0,0 +1,489 @@
+    if (result === 'ok') {
+      updateSettings();
+      message.success(t('notification.SuccessfullyUpdated'));
+    }
```
Let's add error handling.

agatha197 reviewed (changes requested):
1. It's better to set default scheduler type (FIFO)
2. Let's change the enabled checkbox color to token.colorPrimary.
- other checkboxes
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/7d184094-12c4-4e93-9499-cdebd59591cd.png)
- automatically set checkboxes
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/0abd0d00-c241-45c4-b9b4-c3a2efe602c9.png)

nowgnuesLee commented:
> 1. It's better to set default scheduler type (FIFO)
> 2. Let's change the enabled checkbox color to token.colorPrimary.
> - other checkboxes
> ![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/7d184094-12c4-4e93-9499-cdebd59591cd.png)
> - automatically set checkboxes
> ![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/0abd0d00-c241-45c4-b9b4-c3a2efe602c9.png)
> 
> 

This checkbox is a disabled checkbox. The only property that can be modified when it is disabled is colorBgContainerDisabled, but changing this color doesn’t look very good. It’s better to leave it as is. 
I have modified the default value of the scheduler.

yomybaby reviewed (changes requested):
I added a commit to reorganize the order and grouping of setting items. Please also check the following bugs:
- After checking "Unset" and saving, reopening the modal does not show "Unset" checked.
- Changed options in Scheduler setting and saved, but always FIFO is selected.

yomybaby reviewed (changes requested):
- The concerns of the modal should be handled within the modal itself (utilize confirmLoading, okText, onOk, etc.).
- When the save button is clicked in the modal, it should only close on success and should not close on failure.
- Instead of using FormItemWithCheckbox, use a separate Form.Item with CheckBox.

agatha197 reviewed (changes requested):
If there are no changes and you press the 'Display Only Changes' button, add a message such as "There are no changes." along with the Empty component. 

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/65621e82-5d58-4b05-bc5c-355092c0cb66.png)

yomybaby reviewed (approved):
LGTM

nowgnuesLee commented:
> If there are no changes and you press the 'Display Only Changes' button, add a message such as "There are no changes." along with the Empty component. Since this PR is not your concern, feel free to work on it in a separate stack.
> 
> ![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/2HueYSdFvL8pOB5mgrUQ/65621e82-5d58-4b05-bc5c-355092c0cb66.png)
> 
> 

I've created a JIRA issue related to this problem. Since it doesn't seem to be an issue that needs to be resolved in this PR, I'll leave a re-request. [related issue](https://lablup.atlassian.net/browse/FR-769)
